src/helpers.js

```diff
@@ -20,7 +20,13 @@ export const execShellCommand = (cmd) => {
   core.debug(`Executing shell command: [${cmd}]`)
   return new Promise((resolve, reject) => {
     const proc = process.platform !== "win32" ?
-      spawn(cmd, [], { shell: true }) :
+      spawn(cmd, [], {
+        shell: true,
+        env: {
+          ...process.env,
+          HOMEBREW_GITHUB_API_TOKEN: core.getInput('github-token')
```

Comment:
```suggestion
          ...(core.getInput('github-token') ? { HOMEBREW_GITHUB_API_TOKEN: core.getInput('github-token') } : undefined)
```
was thinking about that maybe.

Reply:
I was under the impression that `spawn()` ignores `env` keys with undefined values? Let me try this.

Reply:
we are spreading with undefined, which means that nothing gets spreaded inside the existing ...process.env object. aka. a noop.

Reply:
Yep, if the `env` object contains a key whose value is undefined, it isn't set:

```console
$ node -e "const { spawn } = require('child_process'); const proc = spawn('set -x; env | grep HOMEBREW_GITHUB_API_TOKEN >&2', { shell: true, env: { HOMEBREW_GITHUB_API_TOKEN: undefined } }); proc.stdout.on('data', (data) => { process.stdout.write(data); }); proc.stderr.on('data', (data) => { process.stderr.write(data); });"
+ grep HOMEBREW_GITHUB_API_TOKEN
+ env
```

However, I guess the problem here is that `core.getInput()` would maybe always return a string? How about this instead, then?
```suggestion
          HOMEBREW_GITHUB_API_TOKEN: core.getInput('github-token') || undefined
```

Reply:
[Force-pushed](https://github.com/mxschmitt/action-tmate/compare/b4f2b23d079de54de4b0a7c935a714907765ab8c..0172c1db51e41d8bec70737a72d498f8db394173).